salt/modules/zypperpkg.py

```diff
@@ -218,6 +222,22 @@ def stderr(self):
     def pid(self):
         return self.__call_result.get('pid', '')
 
+    def calc_zypper(self):
+        try:
+            zypp_tmp = version('zypper')
+            # zypper version 1.11.34 in SLE12 update supports vendor change for only dist upgrade
+            if version_cmp(zypp_tmp, '1.11.34') >= 0:
+                # zypper version supports vendor change for dist upgrade
+                self.dup_avc = True
+            # zypper version 1.14.8 in SLE15 update supports vendor change in install/patch/upgrading
+            if version_cmp(zypp_tmp, '1.14.8') >= 0:
```

Comment:
I think we should use a better naming here:

```suggestion
    def refresh_zypper_flags(self):
        try:
            zypp_version = version('zypper')
            # zypper version 1.11.34 in SLE12 update supports vendor change for only dist upgrade
            if version_cmp(zypp_version, '1.11.34') >= 0:
                # zypper version supports vendor change for dist upgrade
                self.dup_avc = True
            # zypper version 1.14.8 in SLE15 update supports vendor change in install/patch/upgrading
            if version_cmp(zypp_version, '1.14.8') >= 0:
```